internal/backends/python/requirements.go

```diff
@@ -0,0 +1,194 @@
+	handle, err := os.Open(path)
+	if err != nil {
+		return []PipFlag{}, sofar, constraints, err
+	}
+	defer handle.Close()
+
+	scanner := bufio.NewScanner(handle)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// Separate out comments
+		segments := strings.SplitN(line, "#", 1)
+		if len(segments) == 2 {
+			line = strings.TrimSpace(segments[0])
+			// comment = strings.TrimSpace(segments[1])
+		}
+
+		if line == "" {
+			// Skip blank lines
+		} else if name, spec, found := findPackage(line); found {
+			// Found a package!
+			sofar[*name] = *spec
+		} else if nextfile, found := strings.CutPrefix(line, "-r "); found {
+			// # It is possible to refer to other requirement files...
+			// -r other-requirements.txt
+			nextfile = filepath.Join(filepath.Dir(path), strings.TrimSpace(nextfile))
+			var newFlags []PipFlag
+			newFlags, sofar, constraints, err = recurseRequirementsTxt(depth+1, nextfile, sofar, constraints)
+			if err != nil {
+				return []PipFlag{}, sofar, constraints, err
+			}
+			flags = append(flags, newFlags...)
+		} else if _, found := strings.CutPrefix(line, "-c "); found {
+			// ... or constraints files.
+			// -c constraints.txt
+			//
+			// TODO: Accumulate constraints to pass to underlyling pip
+		} else if parts := strings.SplitN(line, " ", 2); len(parts) > 1 && knownFlags[parts[0]] {
+			flags = append(flags, PipFlag(strings.Join(parts, " ")))
+			// If we find an editable package, try to extract the package name out of the URI
+			// This both shows that package in the `upm list` output, as well as prevents upm
+			// from inserting it into requirements.txt, which would then cause a conflict
+			// on the next run.
+			if parts[0] == "-e" || parts[0] == "--editable" {
+				matches := matchEggComponent.FindSubmatch([]byte(line))
+				if len(matches) > 1 {
+					sofar[api.PkgName(string(matches[1]))] = ""
+				}
+			}
+		}
+	}
+
+	return flags, sofar, constraints, nil
+}
+
+func ListRequirementsTxt(path string) ([]PipFlag, map[api.PkgName]api.PkgSpec, error) {
+	flags, result, _, err := recurseRequirementsTxt(0, path, make(map[api.PkgName]api.PkgSpec), make(Constraints))
+	return flags, result, err
+}
+
+func recurseRemoveFromRequirementsTxt(depth int, path string, pkgs map[api.PkgName]bool) error {
+	if depth > 10 {
+		util.Die("Too many -r redirects in %s", path)
+	}
+
+	var lines []string
+
+	handle, err := os.OpenFile(path, os.O_RDWR, 0o644)
+	if err != nil {
+		return err
+	}
+	defer handle.Close()
+
+	{
+		scanner := bufio.NewScanner(handle)
+		for scanner.Scan() {
+			line := strings.TrimSpace(scanner.Text())
+
+			if name, _, found := findPackage(line); found && pkgs[normalizePackageName(*name)] {
+				continue
+			} else if nextfile, found := strings.CutPrefix(line, "-r "); found {
+				err := recurseRemoveFromRequirementsTxt(depth+1, nextfile, pkgs)
+				if err != nil {
+					return err
+				}
+				lines = append(lines, line)
+			} else {
+				lines = append(lines, line)
+			}
+		}
+	}
```

Comment:
kinda weird to do anonymous blocks in go
```suggestion
	for scanner := bufio.NewScanner(handle); scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())

		if name, _, found := findPackage(line); found && pkgs[normalizePackageName(*name)] {
			continue
		} else if nextfile, found := strings.CutPrefix(line, "-r "); found {
			err := recurseRemoveFromRequirementsTxt(depth+1, nextfile, pkgs)
			if err != nil {
				return err
			}
			lines = append(lines, line)
		} else {
			lines = append(lines, line)
		}
	}
```